### PR TITLE
fix: updated general docs and format

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR improves the overall documentation of the generated source code as well as the template documentation.

Changes: 
* Moved some doc files to `./docs`
* Added hook to move  `./docs` files into the output
* Added JSDoc to all generated code 
* Renamed some variables to make more sense
* Fixed typos

**Related issue(s)**
Solves https://github.com/asyncapi/ts-nats-template/issues/8
Solves https://github.com/asyncapi/ts-nats-template/issues/7

### CODE REVIEW

File: template/src/channels/$$channel$$.ts.js [hunk 1]
```diff
@@ -70,6 +70,11 @@ export default function channelRender({ asyncapi, channelName, channel, params }
     {`
 ${General(channel, publishMessage, subscribeMessage, '..')}
 
+/**
+ * Module which wraps functionality for the channel ${channelName}
```
Comment (magicmatatjahu) [inline template/src/channels/$$channel$$.ts.js:74]:
```suggestion
 * Module which wraps functionality for the `${channelName}` channel
```

File: template/src/testclient/testchannels/$$channel$$.ts.js [hunk 2]
```diff
@@ -67,6 +67,10 @@ export default function channelRender({ asyncapi, channelName, channel, params }
     {`
 ${General(channel, publishMessage, subscribeMessage, '../..')}
 
+/**
+ * Module which wraps functionality for the channel ${channelName}
```
Comment (magicmatatjahu) [inline template/src/testclient/testchannels/$$channel$$.ts.js:71]:
```suggestion
 * Module which wraps functionality for the `${channelName}` channel
```

**Review by magicmatatjahu — COMMENTED** [review]
Very good! Only some minor suggestions for some descriptions in jsDoc

File: README.md [hunk 3]
```diff
@@ -51,207 +54,18 @@ async function connect(){
...
-
-### String payloads
-If you only have 1 hook then you must ensure that it returns a string.
+# Features
+* Supports wildcard channels. AsyncAPI describes the channel path to be defined as [RFC 6570 URI](https://www.asyncapi.com/docs/specifications/2.0.0/#a-name-channelsobject-a-channels-object). So a channel containing a wildcard needs to be defined with parameters such as `smartylighting/streetlights/{wildcard}`.
+* Supports [test/mirror client](#test-client) for testing or other useful scenarios.
+* This template can be used as a NodeJS library.
+* Generates payload models using the [AsyncAPI model generation library](https://github.com/asyncapi/generator-model-sdk). 
+# Restrictions 
+* Empty objects are not supported, use `null` types instead.
```
[the remark this answers is not in the document]
Comment (jonaslagoni) [reply README.md:70]: Without knowing 100% I think the restriction no longer applies. It is on the list of things I have to check. Cause when we switched from quicktype to our model generation, I also think these issues were resolved.

File: docs/events.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
+# Available events
+The generated client can emit multiple events, all of these are re-emitted from the [nats-ts](https://github.com/nats-io/nats.ts#notifications) library. 
+
+In the following examples `nats-client` is the generated client wrapper.
+
+## TypeScript example
```
[the remark this answers is not in the document]
Comment (jonaslagoni) [reply docs/events.md:6]: Hmm, depends right. I know that when I was using a typescript library and when Typescript was not really something I used I loved when they provided NodeJS examples 😄

[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply docs/events.md:6]: In some situation, like using generic arguments, interfaces etc is good, but here is only typings for function parameters.

[the remark this answers is not in the document]
Comment (jonaslagoni) [reply docs/events.md:6]: Yep, but it just makes it easier to pick and choose between the events 😄 I will keep them for now, but keep it in mind :)

**Review by magicmatatjahu — APPROVED** [review]

**Pull request merged**